kasa/smart/smartchilddevice.py

```diff
@@ -49,6 +50,20 @@ def __init__(
         self._update_internal_state(info)
         self._components = component_info
 
+    @property
+    def _device_info(self) -> _DeviceInfo:
+        """Return device info."""
+        components = [
+            {"id": id, "ver_code": ver} for id, ver in self._components.items()
+        ]
```

Comment:
Should this be located in `smartdevice.py` instead?

Reply:
This is only applicable to the `smartchilddevice` because it doesn't have the `components_raw`. A better way would probably be to pass in the `components_raw` and have the smart child device do the conversion to dict itself once. What do you think?

Reply:
Now resolved after https://github.com/python-kasa/python-kasa/pull/1363